src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/ApprenticeController.cs

```diff
@@ -544,15 +547,78 @@ public async Task<IActionResult> EditApprenticeship(EditApprenticeshipRequestVie
         {
             var validationRequest = await _modelMapper.Map<ValidateApprenticeshipForEditRequest>(viewModel);
             await _commitmentsApiClient.ValidateApprenticeshipForEdit(validationRequest);
-
-            return RedirectToAction("ConfirmEditApprenticeship", new { viewModel.AccountHashedId, ApprenticeshipHashedId = viewModel.HashedApprenticeshipId });
+         
+            TempData.Put("EditApprenticeshipRequestViewModel", viewModel);
+            return RedirectToAction("ConfirmEditApprenticeship", new { apprenticeshipHashedId = viewModel.HashedApprenticeshipId, accountHashedId = viewModel.AccountHashedId });
         }
 
         [HttpGet]
         [Route("{apprenticeshipHashedId}/edit/confirm")]
-        public IActionResult ConfirmEditApprenticeship(ConfirmEditApprenticeshipRequest request)
+        public async Task<IActionResult> ConfirmEditApprenticeship()
         {
-            return View();
+            var editApprenticeshipRequestViewModel = TempData.GetButDontRemove<EditApprenticeshipRequestViewModel>("EditApprenticeshipRequestViewModel");
+            var viewModel = await _modelMapper.Map<ConfirmEditApprenticeshipViewModel>(editApprenticeshipRequestViewModel);
+
+            return View(viewModel);
+        }
+
+        [HttpPost]
+        [Route("{apprenticeshipHashedId}/edit/confirm")]
+        public async Task<IActionResult> ConfirmEditApprenticeship(ConfirmEditApprenticeshipViewModel viewModel)
+        {
+            if (viewModel.ConfirmChanges.Value)
+            {
+                var request = await _modelMapper.Map<EditApprenticeshipApiRequest>(viewModel);
+                var result = await _commitmentsApiClient.EditApprenticeship(request);
+
+                if (result.NeedReapproval)
+                {
+                    TempData.AddFlashMessage(ApprenticeChangesSentToProvider, ITempDataDictionaryExtensions.FlashMessageLevel.Info);
+                }
+                else
+                {
+                    TempData.AddFlashMessage(ApprenticeUpdated, ITempDataDictionaryExtensions.FlashMessageLevel.Info);
+                }
+            }
+
+            return await Task.FromResult(RedirectToAction(nameof(ApprenticeshipDetails), new { viewModel.AccountHashedId, viewModel.ApprenticeshipHashedId }));
+        }
+
+        [HttpGet]
+        [Route("{apprenticeshipHashedId}/changes/review")]
+        public async Task<IActionResult> ReviewApprenticeshipUpdates(ReviewApprenticehipUpdatesRequest request)
+        {
+            var viewModel = await _modelMapper.Map<ReviewApprenticeshipUpdatesRequestViewModel>(request);
+
+            return View(viewModel);
+        }
+
+        [HttpPost]
+        [Route("{apprenticeshipHashedId}/changes/review")]
+        public async Task<IActionResult> ReviewApprenticeshipUpdates(ReviewApprenticeshipUpdatesRequestViewModel viewModel)
+        {
+            if (viewModel.ApproveChanges.Value)

```

Comment:
why is this Nullable<bool>? If this is null then this controller action will throw an exception